src/Makefile.am

```diff
@@ -51,3 +53,27 @@ nobase_include_HEADERS = \
 
 nobase_noinst_HEADERS = \
         ../include/opendht/request.h
+
+######################
+#  ARGON2 submodule  #
+######################
+
+noinst_LTLIBRARIES = libargon2.la
```

Comment:
> - built even if system argon2 is used (check with WITH_INCLUDED_ARGON2 macro)

Still on it. I'll update when done.

> - dependency issue: bug can be reproduced easily without parallel build: make clean and make (without -j)
>> ../libtool: line 7485: cd: ../argon2/src/.libs: No such file or directory
>> libtool: error: cannot determine absolute directory name of '../argon2/src/.libs'

I don't experience this on 4646756. Logically, this should not happen because of the first point you mentionned. The path `../argon2/src/.libs` should always be found. Are you sure you're on the good revision?